google-account-plugin/src/com/google/cloud/tools/intellij/login/ui/GoogleLoginEmptyPanel.java

```diff
@@ -53,14 +50,14 @@ public GoogleLoginEmptyPanel() {
 
     contentScrollPane = new JBScrollPane();
     JButton addAccountButton = new JButton(needsToSignIn() ? SIGN_IN : ADD_ACCOUNT);
-    addAccountButton.addActionListener(new ActionListener() {
-      @Override
-      public void actionPerformed(ActionEvent event) {
-        doLogin();
-      }
-    });
+    addAccountButton.addActionListener(event -> doLogin());
     addAccountButton.setHorizontalAlignment(SwingConstants.LEFT);
 
+    ProjectUtil.getInstance()
```

Comment:
Won't this leave a bunch of dangling subscribers for buttons that no longer exist?

Reply:
yea. this needs some cleanup

Reply:
a similar problem is that the subscribers keep getting registered multiple times (every time the project selector is initialized)